src/native/credentials_provider.c

```diff
@@ -654,7 +666,9 @@ struct aws_credentials_provider_get_credentials_callback_data {
 };
 
 static void s_cp_callback_data_clean_up(struct aws_credentials_provider_get_credentials_callback_data *callback_data) {
-    JNIEnv *env = aws_jni_get_thread_env(callback_data->jvm);
+
+    /********** JNI ENV ACQUIRE **********/
+    JNIEnv *env = aws_jni_acquire_thread_env(callback_data->jvm);
```

Comment:
I still think `env` should just be passed to this function as a param.
This is a helper function, called from 2 places, both of which already have `env` available